### ISSUE TITLE
Problem for some Chinese users to connect

### ISSUE THREAD

**Issue by jonjamcam** [elieserdejesus/JamTaba#821]
I was talking to RP about China users having problems to connect not only to Jamtaba, but to ninjam in general.

The most common issue is disconnection when using other clients like wahjam after being able to reach a server.

In the case of Jamtaba, the problem is mostly no connection. He sent me this snap:

![image](https://cloud.githubusercontent.com/assets/15310433/25012049/aa795bd2-2045-11e7-9f5e-5c0f157a2d14.png)

 Some China users, RP tells me, are already using proxy servers to connect, but the disconnection problem persists.

**Comment by elieserdejesus** [comment]
@jonjamcam , in general "Connection time out" errors can be related with very slow connections. I think the default time out is 30 seconds. JamTaba is trying connect in JTBA server, after 30 seconds without any network response a "time out" error will be generated. In general proxies are slow, and my guess is: proxies + slow connections are not a good combination to play online.

As you know netowrk policies in China are very "interesting" :)  I suspect China internet connections are very slow. Proxies can solve the the "government filters", but the cables/wires are still poor. I think this problem can not be solved at software level.

**Comment by jonjamcam** [comment]
> I think this problem can not be solved at software level.

I agree there's no definite solution in this case. But if someone managed to get a successful connection using a proxy, then Jamtaba can't use it unless you use special software that sends all your connections through the proxy.

I did some experiments last year using QNetworkProxy (http://doc.qt.io/qt-5/qnetworkproxy.html) and managed to connect Jamtaba very well to several proxies. The ninjam connection not only worked OK, but I was able to play smoothly for quite some time. 

Just an idea to consider for future versions. Maybe this can help a little.
